.github/workflows/indexer-processor-testing.yaml

```diff
@@ -0,0 +1,144 @@
+
+name: Trigger Processor Tests on JSON Change
+
+on:
+  workflow_dispatch:
+  pull_request:  # Trigger on PR-level events
+    branches:
+      - main
+
+# the required permissions to request the ID token
+permissions:
+  id-token: write  # This is required for GCP authentication
+  contents: read   # Ensure the workflow has access to repository contents
+
+jobs:
+  dispatch_event:
+    runs-on: runs-on,cpu=64,family=c7,hdd=500,image=aptos-ubuntu-x64,run-id=${{ github.run_id }}
+
+    steps:
+      - name: Checkout the repository
+        uses: actions/checkout@v4
+        with:
+          ref: ${{ github.event.pull_request.head.ref }}
+
+      - name: Set up Rust
+        uses: aptos-labs/aptos-core/.github/actions/rust-setup@main
+        with:
+          GIT_CREDENTIALS: ${{ secrets.GIT_CREDENTIALS }}
+
+      # Install necessary system dependencies
+      - name: Install system dependencies
+        run: |
+          sudo apt-get update
+          sudo apt-get install build-essential libssl-dev pkg-config
+
+      # Ensure Rust is updated
+      - name: Update Rust toolchain
+        run: rustup update
+
+      - name: Run CLI to Generate JSON Files
+        run: |
+          cd ecosystem/indexer-grpc/indexer-transaction-generator
+          cargo run -- --testing-folder ./example_tests --output-folder ../indexer-test-transactions/new_json_transactions
+
+      - name: Install jq
+        run: sudo apt-get install jq  # Ensure jq is installed for JSON processing
+
+      # TODO: improve this step to be easily maintainable and extensible
+      # Prepare Original and New JSON Files
+      - name: Prepare and Clean JSON Files
+        run: |
+          cd ecosystem/indexer-grpc/indexer-test-transactions
+
+          for folder in json_transactions/scripted_transactions new_json_transactions/scripted_transactions; do
+            for file in $folder/*.json; do
+              echo "Processing $file..."
+              base_file=$(basename "$file")
+
+              jq 'del(.timestamp,
+                      .version,
+                      .info.hash,
+                      .info.stateChangeHash,
+                      .info.accumulatorRootHash,
+                      .info.changes[].writeResource.stateKeyHash,
+                      .info.changes[].writeResource.type.address,
+                      .info.changes[].writeResource.address,
```

Comment:
These should not change right?

Reply:
these could also change when the txn gets newly generated. cc @larry-aptos  to confirm

Reply:
Ohh if it's because we create different object addresses in between txn generations, that makes sense!

Reply:
transactions content can change if we regenerate from localnet. 

i think instead of removing, we should have a tool that can diff json with ignoring fields. This tool will be only used to script transactions.

Reply:
yeah we can improve this as we migrate processros  or add more tests cases!